docs/content/latest/api/ysql/the-sql-language/statements/ddl_truncate.md

```diff
@@ -43,52 +45,125 @@ Use the `TRUNCATE` statement to clear all rows in a table.
   </div>
 </div>
 
-## Semantics
-
-### *truncate*
-
-#### TRUNCATE [ TABLE ] { { [ ONLY ] *name* [ * ] } [ , ... ] }
+{{< note title="Table inheritance is not yet supported" >}}
+The [table_expr](../../../syntax_resources/grammar_diagrams/#table-expr) rule specifies syntax that is useful only when at least one other table inherits one of the tables that the `truncate` statement lists explicitly. See [this note](../ddl_alter_table#table-expr-note) for more detail. Until inheritance is supported, use a bare [table_name](../../../syntax_resources/grammar_diagrams/#table-name).
+{{< /note >}}
 
-#### *name*
+## Semantics
 
 Specify the name of the table to be truncated.
 
 - `TRUNCATE` acquires `ACCESS EXCLUSIVE` lock on the tables to be truncated. The `ACCESS EXCLUSIVE` locking option is not yet fully supported.
 - `TRUNCATE` is not supported for foreign tables.
+- `CASCADE` and `RESTRICT` affect what happens when the table that the `TRUNCATE` statement targets has dependent tables. A dependent table (and, in turn, its dependent tables) have that status because they have direct or transitive foreign key constrains to the target table. `CASCADE` causes the closure of dependent tables all to be truncated. And `RESTRICT` causes the `TRUNCATE` attempt to fail if the target table has any dependent tables. This error outcome is the same even when all of the tables are empty. If neither `CASCADE` nor `RESTRICT` is written, then the effect is as if `RESTRICT` had been written.
 
-## Examples
 
-```plpgsql
-yugabyte=# CREATE TABLE sample(k1 int, k2 int, v1 int, v2 text, PRIMARY KEY (k1, k2));
-```
+## Example
 
-```plpgsql
-yugabyte=# INSERT INTO sample VALUES (1, 2.0, 3, 'a'), (2, 3.0, 4, 'b'), (3, 4.0, 5, 'c');
-```
+First create a parent-child table pair and populate them:
 
 ```plpgsql
-yugabyte=# SELECT * FROM sample ORDER BY k1;
+drop table if exists children cascade;
+drop table if exists parents  cascade;
+
+create table parents(k int primary key, v text not null);
+
+create table children(
+  parent_k  int  not null,
+  k         int  not null,
+  v         text not null,
+
+  constraint children_pk primary key(parent_k, k),
+
+  constraint children_fk foreign key(parent_k)
+    references parents(k)
+    match full
+    on delete cascade
+    on update restrict);
+
+insert into parents(k, v) values (1, 'dog'), (2, 'cat'), (3, 'frog');
+
+insert into children(parent_k, k, v) values
+  (1, 1, 'dog-child-a'),
+  (1, 2, 'dog-child-b'),
+  (1, 3, 'dog-child-c'),
+  (2, 1, 'cat-child-a'),
+  (2, 2, 'cat-child-b'),
+  (2, 3, 'cat-child-c'),
+  (3, 1, 'frog-child-a'),
+  (3, 2, 'frog-child-b'),
+  (3, 3, 'frog-child-c');
+
+select p.v as "parents.v", c.v as "chilgren.v"
```

Comment:
Chilgren?

Reply:
Good catch. The Typora spell checker didn't catch this. (Maybe 'cos of the embedded period.) Fixed locally.